src/java.base/share/classes/java/io/Reader.java

```diff
@@ -207,6 +208,11 @@ public int read(char[] cbuf, int off, int len) throws IOException {
                 return n;
             }
 
+            public String readAllChars() throws IOException {
```

Comment:
Now that the method got renamed from `readString()` to `readAllChars()` I'd like to suggest that the return type should not be `String` anymore but `CharSequence`. This would open potential for more efficient implementations. For example, we can omit the final `.toString()` after `.subSequence()`, effectively preventing the *double* copy implied (the first copy already was created by `subSequence()`). *Iff* a caller essentially needs a `String` (for whatever reason), he can add a `.toString()` *in the calling code* easily. But unless that is the case, we should not *enforce* `String` for efficiency reasons.